### PR TITLE
Sealed storage peer

### PR DESCRIPTION
**Pull request by mjtognetti**
Encrypt workspace urls when writing into the storage peer registry.

TODO:
- [x] finalize storage peer changes and reflect those changes here
- [x] clean up UI (better messaging around registration status)
- [x] update hypermerge dependency to depend on mainline.

### CODE REVIEW

File: src/renderer/components/content-types/storage-peer/index.ts [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
 
 import StoragePeerWorkspace from './StoragePeerWorkspace'
 import StoragePeer from './StoragePeer'
 import { DeviceDoc } from '../workspace/Device'
 
 export interface StoragePeerDoc extends DeviceDoc {
-  storedUrls: { [contact: string /* HypermergeUrl */]: HypermergeUrl }
+  publicKey: Crypto.EncodedPublicEncryptionKey
+  publicKeySignature: Crypto.EncodedSignature
+  registry: { [contact: string /* HypermergeUrl */]: Crypto.EncodedSealedBox }
```
Comment (mjtognetti) [inline src/renderer/components/content-types/storage-peer/index.ts]: Going to wipe the contact information from here.

Comment (mjtognetti) [reply src/renderer/components/content-types/storage-peer/index.ts]: Nevermind! We decided to keep this. I've added UI to make this knowledge explicit.

File: src/renderer/components/content-types/storage-peer/StoragePeerHooks.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  // The encryptionKey and signature may be missing from the doc.
+  if (!doc.encryptionKey || !doc.encryptionKeySignature) return null
+  const isValid = repo.crypto.verify(docUrl, doc.encryptionKey, doc.encryptionKeySignature)
+  return isValid ? doc.encryptionKey : null
+}
+
+/**
+ * Helper function for removing an item from an Automerge list.
+ */
+function without<T>(val: T, list: Automerge.List<T>) {
```
Comment (pvh) [inline src/renderer/components/content-types/storage-peer/StoragePeerHooks.ts]: i like this. we should upstream it.

File: package.json [hunk 3]
```diff
@@ -40,7 +40,7 @@
     "electron-squirrel-startup": "^1.0.0",
     "freeze-dry": "^0.2.4",
     "html-webpack-plugin": "^3.2.0",
-    "hypermerge": "github:automerge/hypermerge",
+    "hypermerge": "github:automerge/hypermerge#sealed-box",
```
Comment (pvh) [inline package.json]: remember to either file a follow-up to switch back or else mainline

File: src/renderer/components/content-types/storage-peer/StoragePeer.tsx [hunk 4]
```diff
@@ -2,32 +2,29 @@ import React from 'react'
...
 
   const { context, url, hypermergeUrl } = props
-  const { name, storedUrls } = doc
+  const { name } = doc
 
   const title = name
-  const subtitle = `${storedUrls.length} URL${
-    Object.keys(storedUrls).length === 1 ? '' : 's'
-  } stored`
+  const subtitle = isRegistered ? 'registered' : 'not registered'
```
Comment (pvh) [inline src/renderer/components/content-types/storage-peer/StoragePeer.tsx]: registered vs not registered feels like the wrong term here.

Comment (mjtognetti) [reply src/renderer/components/content-types/storage-peer/StoragePeer.tsx]: Agreed - don't love it. Previously we said "Register with the Storage Peer" and then had a "Stored Workspaces" list. Any thoughts on better terminology.

**Comment by pvh** [comment]
The other thing we discussed (and I think agreed?) was that we should document what the Storage Peer does and what the caveats are for someone using it.

**Pull request merged**